diff --git a/diff.module b/diff.module
index d3050e3..b26d57a 100644
--- a/diff.module
+++ b/diff.module
@@ -265,8 +265,9 @@ function diff_entity_info_alter(&$entity_info) {
 /**
  * Returns a list of all the existing revision numbers.
  *
- * Clone of node_revision_list() with revision status included. This would be
- * an additional join in Drupal 8.x to the {node_field_revision} table.
+ * Clone of node_revision_list() with revision status included, and sorted by
+ * timestamp. This would be an additional join in Drupal 8.x to the
+ * {node_field_revision} table.
  *
  * @param object $node
  *   The node object.
@@ -276,7 +277,7 @@ function diff_entity_info_alter(&$entity_info) {
  */
 function diff_node_revision_list($node) {
   $revisions = array();
-  $result = db_query('SELECT r.vid, r.title, r.log, r.uid, n.vid AS current_vid, r.status AS status, r.timestamp, u.name FROM {node_revision} r LEFT JOIN {node} n ON n.vid = r.vid INNER JOIN {users} u ON u.uid = r.uid WHERE r.nid = :nid ORDER BY r.vid DESC', array(':nid' => $node->nid));
+  $result = db_query('SELECT r.vid, r.title, r.log, r.uid, n.vid AS current_vid, r.status AS status, r.timestamp, u.name FROM {node_revision} r LEFT JOIN {node} n ON n.vid = r.vid INNER JOIN {users} u ON u.uid = r.uid WHERE r.nid = :nid ORDER BY r.timestamp DESC, r.vid DESC', array(':nid' => $node->nid));
   foreach ($result as $revision) {
     $revisions[$revision->vid] = $revision;
   }
@@ -352,7 +353,7 @@ function diff_node_view_alter(&$build) {
     // Ugly but cheap way to check that we are viewing a node's revision page.
     if (arg(2) === 'revisions' && arg(3) === $node->vid) {
       module_load_include('inc', 'diff', 'diff.pages');
-      $old_vid = _diff_get_previous_vid(node_revision_list($node), $node->vid);
+      $old_vid = _diff_get_previous_vid(diff_node_revision_list($node), $node->vid);
       $build = array('#markup' => diff_inline_show($node, $old_vid));
     }
     $build['#prefix'] = isset($build['#prefix']) ? "<div id='diff-inline-{$node->nid}'>" . $build['#prefix'] : "<div id='diff-inline-{$node->nid}'>";
diff --git a/diff.pages.inc b/diff.pages.inc
index 285eb1e..9a586e5 100644
--- a/diff.pages.inc
+++ b/diff.pages.inc
@@ -9,7 +9,7 @@
  * Menu callback - show latest diff for a given node.
  */
 function diff_latest($node) {
-  $revisions = node_revision_list($node);
+  $revisions = diff_node_revision_list($node);
   if (count($revisions) < 2 || !diff_node_revision_access($node, 'view')) {
     drupal_goto('node/' . $node->nid);
   }
@@ -32,7 +32,7 @@ function diff_diffs_overview($node) {
 /**
  * Input form to select two revisions.
  */
-function diff_node_revisions($form, $form_state, $node) {
+function diff_node_revisions($form, &$form_state, $node) {
   $form['nid'] = array(
     '#type' => 'hidden',
     '#value' => $node->nid,
@@ -58,6 +58,8 @@ function diff_node_revisions($form, $form_state, $node) {
     $revisions = $revision_list;
   }
 
+  $form_state['revisions'] = $revisions;
+
   $revert_permission = FALSE;
   if ((user_access('revert revisions') || user_access('administer nodes')) && node_access('update', $node)) {
     $revert_permission = TRUE;
@@ -146,8 +148,15 @@ function diff_node_revisions_validate($form, &$form_state) {
  */
 function diff_node_revisions_submit($form, &$form_state) {
   // The ids are ordered so the old revision is always on the left.
-  $old_vid = min($form_state['values']['old'], $form_state['values']['new']);
-  $new_vid = max($form_state['values']['old'], $form_state['values']['new']);
+  $revisions = $form_state['revisions'];
+  if ($revisions[$form_state['values']['old']]->timestamp > $revisions[$form_state['values']['new']]->timestamp) {
+    $old_vid = $form_state['values']['new'];
+    $new_vid = $form_state['values']['old'];
+  }
+  else {
+    $old_vid = $form_state['values']['old'];
+    $new_vid = $form_state['values']['new'];
+  }
   if (isset($_GET['destination'])) {
     unset($_GET['destination']);
   }
@@ -181,7 +190,7 @@ function diff_diffs_show($node, $old_vid, $new_vid, $state = NULL) {
   if (node_is_page($node)) {
     drupal_set_title(t('Revisions for %title', array('%title' => $node->title)), PASS_THROUGH);
   }
-  $node_revisions = node_revision_list($node);
+  $node_revisions = diff_node_revision_list($node);
 
   $old_node = node_load($node->nid, $old_vid);
   $new_node = node_load($node->nid, $new_vid);
@@ -652,7 +661,7 @@ function diff_inline_show($node, $vid = 0, $metadata = TRUE) {
   node_build_content($new_node);
   $new = drupal_render($new_node->content);
 
-  $old = $vid ? _diff_get_previous_vid(node_revision_list($node), $vid) : 0;
+  $old = $vid ? _diff_get_previous_vid(diff_node_revision_list($node), $vid) : 0;
   if ($old) {
     $old_node = node_load($node->nid, $old);
     node_build_content($old_node);
